src/main/drivers/max7456.c

```diff
@@ -571,14 +571,18 @@ uint8_t* max7456GetScreenBuffer(void)
 
 void max7456WriteChar(uint8_t x, uint8_t y, uint8_t c)
 {
-    screenBuffer[y*CHARS_PER_LINE+x] = c;
+    if (y < VIDEO_LINES_PAL) {
+        screenBuffer[y*CHARS_PER_LINE+x] = c;
+    }
```

Comment:
This can still overflow the buffer. For example if y is the max row and x >= CHARS_PER_LINE. Probably best to calculate the index and then compare that to `VIDEO_BUFFER_CHARS_PAL`. Also to be consistent with the string write it should not allow `x >= CHARS_PER_LINE` (which would make the character wrap to the next line).

Reply:
Thanks for the feedback. I think as long as we control the horizontal, and the vertical, we can skip calculating the index.